src/crm/mailjet.ts

```diff
@@ -73,78 +66,121 @@ class Mailjet extends CRM {
+        });
+
+      return true;
+    } catch (error: any) {
+      console.error(`Failed to add contact ${email} to list ${this.list}: ${error.statusCode}`);
+      return false;
+    }
+  }
 
-const source = "proca";
-const camp = {id: 0, name: "todo"};
+  updateContactProperties = async (message: ActionMessage): Promise<boolean> => {
 
-const action = message.action;
+    const data = Object.entries(this.contactProperties).map(([sourceField, mailjetField]) => ({
+        Name: mailjetField,
+      Value: message.contact[sourceField] || ""
+      }));
 
-    if (existing === false) {
-      return this.createContact(message.contact, action, camp.id, source);
-    } else {
-console.log(existing);
-      return this.updateContact(existing, message.contact, action, camp.id, source);
+    try {
+      await this.mailjet
+        .put("contactdata", { 'version': 'v3' })
+        .id(message.contact.email)
+        .request({
+          "Data": data
+        })
+
+      return true;
+    } catch (e) {
+      console.error(`Updating properties for contact ${message.contact.email}, action ${message.actionId} faile with ${e.statusCode}`);
+      return false;
+	    }
     }
-    return false;
-} catch (e:any) {
-  console.log(e); throw new Error (e);
-}
-}
 
-  createContact = async (
-    contact: any,
-    action: any,
-    campaign_id: number,
-    source?: string
-  ): Promise<boolean> => {
+  handleContact = async (
+    message: ActionMessage
+  ): Promise<handleResult | boolean> => {
 
-const request = await this.mailjet
-	.post("contact", {'version': 'v3'})
-	.request({
-      "Name":contact.lastName ? contact.firsttName + ' ' + contact.lastName : contact.firstName,
-      "Email":contact.email
-    });
+    console.log("Action taken from the queue", message.action.id);
 
-console.log("request", request);
-   return false;
-}
-
-  updateContact = async (
-    crmContact: any,
-    contact: any,
-    action: any,
-    campaign_id: number,
-    source?: string
-  ): Promise<boolean> => {
-    if (campaign_id === null) {
-      throw new Error ("missing campaign id"); 
+    if (this.verbose) {
+      console.log(message);
     }
-   return false;
-}
-
-  fetchCampaign = async (campaign: ProcaCampaign): Promise<any> => {
-console.log(campaign);
 
-     return this.list;
-  };
-  fetchContact = async (email: string, context: any): Promise<any> => {
-    const result= await this.mailjet.get("contact",{'version': 'v3'}).id(email).request();
-
-  if (result.body.Data.count ===0) 
-    return false;
-  const contact=result.body.Data[0];
-console.log(contact);
-  return true;
+    try {
+      // create contact
+      const { response: { status } } = await this.mailjet
+        .post("contact", { 'version': 'v3' })
+        .request({
+          "Name": message.contact.lastName ? message.contact.firstName + " " + message.contact.lastName : message.contact.firstName,
+          "Email": message.contact.email
+        });
+
+      if (status === 201) {
```

Comment:
I don't think you need to check if status === 201, it does throw an error if there is a problem

Reply:
right